src/psyclone/gocean1p0.py

```diff
@@ -3038,6 +3140,31 @@ def data_on_device(self, parent):
                                          rhs=".true."))
                     obj_list.append(var)
 
+    def lower_to_language_level(self):
+        '''
+        In-place replacement of DSL or high-level concepts into generic
+        PSyIR constructs. In addition to call its parent logic, the
+        GOACCEnterDataDirective sets up the 'data_on_device' flag for
+        each of the fields accessed.
+
+        '''
+        self._acc_dirs = self.ancestor(InvokeSchedule).walk(
+                (ACCParallelDirective, ACCKernelsDirective))
+        obj_list = []
+        for pdir in self._acc_dirs:
+            for var in pdir.fields:
```

Comment:
I just copied the method above (data_on_device) and done the same but with PSyIR instead of f2pygen.

I think data_on_device attribute just exist in the r2d_field of dl_esm_inf (if I recall correctly there is a dl_esm_inf issue to extend a similar interface to grid properties which are arrays but is not done yet)

Reply:
Ah, OK, that makes sense.